macros/sql/dedupe.sql

```diff
@@ -0,0 +1,41 @@
+{%- macro dedupe(table, group_by, order_by=none) -%}
+    {{ return(adapter.dispatch('dedupe', 'dbt_utils')(table, group_by, order_by=order_by)) }}
+{% endmacro %}
+
+{%- macro default__dedupe(table, group_by, order_by=none) -%}
+
+    select
+        {{ dbt_utils.star(table, relation_alias='deduped') | indent }}
+    from (
+        select
+            _inner.*,
+            row_number() over (
+                partition by {{ group_by }}
+                {% if order_by != none -%}
```

Comment:
I can't remember whether `!=` works for `none` - I tend to use `if not order_by` or `if order_by is not none`. If you haven't validated this, it'd be worth double-checking 🙏 . I'm worried that your approach might be the same as `column != null` in SQL which doesn't work. 

Related - dbt Labs should really write a Jinja style guide to make explicit the best way to check for the existence of optional values...